types/map_seq_type.toml

```diff
@@ -102,29 +102,29 @@ type = '''types::st::List<DB, types::st::Pair<DB,
 func = '''
 using element_type = std::pair<T0, T1>;
 
-static constexpr std::array<std::string_view, 1> names{"TODO"};
-
 static constexpr std::array<inst::Field, 2> entryFields{
   inst::Field{
     sizeof(T0),
     "key",
-    names,
+    NameProvider<T0>::names,
     TypeHandler<DB, T0>::fields,
     TypeHandler<DB, T0>::processors,
   },
   inst::Field{
     sizeof(T1),
     "value",
-    names,
+    NameProvider<T1>::names,
     TypeHandler<DB, T1>::fields,
     TypeHandler<DB, T1>::processors,
   },
 };
+
+static constexpr std::array<std::string_view, 1> entryNames{"entry"};
```

Comment:
How will this appear in the final output?

Reply:
It's kind of misleading but to nest them this way it needs a type. I'm wondering now if perhaps an empty array of `typeNames` would be better here, given it doesn't correspond to a C++ type. Open to suggestions or leaving it like this for now and making it better later.

```
[
{
  "name" : "a0",
  "typePath" : ["a0"],
  "typeNames" : ["folly::sorted_vector_map<int32_t, int32_t, less<int>, std::allocator<std::pair<int32_t, int32_t>>, void, std::vector<std::pair<int32_t, int32_t>, std::allocator<std::pair<int32_t, int32_t>>>>"],
  "staticSize": 24,
  "exclusiveSize": 24,
  "pointer": 140736465865216,
  "length": 2,
  "capacity": 2,
  "members": [
    {
      "name" : "[]",
      "typePath" : ["a0", "[]"],
      "typeNames" : ["entry"],
      "staticSize": 8,
      "exclusiveSize": 0,
      "members": [
        {
          "name" : "key",
          "typePath" : ["a0", "[]", "key"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        },
        {
          "name" : "value",
          "typePath" : ["a0", "[]", "value"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        }
        ]
},
    {
      "name" : "[]",
      "typePath" : ["a0", "[]"],
      "typeNames" : ["entry"],
      "staticSize": 8,
      "exclusiveSize": 0,
      "members": [
        {
          "name" : "key",
          "typePath" : ["a0", "[]", "key"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        },
        {
          "name" : "value",
          "typePath" : ["a0", "[]", "value"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        }
        ]
}
    ]
}
```

Reply:
An empty name might be better, since at least people won't start searching for "entry" in our codebase to see what type they're looking at (and it's even more confusing if there actually is a real type called "entry"!).

Reply:
```
[
{
  "name" : "a0",
  "typePath" : ["a0"],
  "typeNames" : ["folly::sorted_vector_map<int32_t, int32_t, less<int>, std::allocator<std::pair<int32_t, int32_t>>, void, std::vector<std::pair<int32_t, int32_t>, std::allocator<std::pair<int32_t, int32_t>>>>"],
  "staticSize": 24,
  "exclusiveSize": 24,
  "pointer": 140721064383744,
  "length": 2,
  "capacity": 2,
  "members": [
    {
      "name" : "[]",
      "typePath" : ["a0", "[]"],
      "typeNames" : [],
      "staticSize": 8,
      "exclusiveSize": 0,
      "members": [
        {
          "name" : "key",
          "typePath" : ["a0", "[]", "key"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        },
        {
          "name" : "value",
          "typePath" : ["a0", "[]", "value"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        }
        ]
},
    {
      "name" : "[]",
      "typePath" : ["a0", "[]"],
      "typeNames" : [],
      "staticSize": 8,
      "exclusiveSize": 0,
      "members": [
        {
          "name" : "key",
          "typePath" : ["a0", "[]", "key"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        },
        {
          "name" : "value",
          "typePath" : ["a0", "[]", "value"],
          "typeNames" : ["int32_t"],
          "staticSize": 4,
          "exclusiveSize": 4,
          "members": []
        }
        ]
}
    ]
}
]
```